src/main/kotlin/org/danilopianini/plagiarismdetector/repository/content/RepoContentSupplierStrategy.kt

```diff
@@ -0,0 +1,13 @@
+package org.danilopianini.plagiarismdetector.repository.content
+
+import java.io.File
+
+/**
+ * An interface modeling the strategy used for retrieve the repository content.
+ */
+interface RepoContentSupplierStrategy {
+    /**
+     * @return the files with the given extensions.
+     */
+    fun getFilesOf(extensions: Iterable<String>): Iterable<File>
```

Comment:
Let us use some more flexibility!
```suggestion
    fun filesMatching(pattern: Regex): Iterable<File>
```

Reply:
Also, use sequences

Reply:
Or Flows